src/evo/deterministicmns.cpp

```diff
@@ -389,7 +397,30 @@ bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, const C
             if (newList.HasUniqueProperty(proTx.keyIDOwner) || newList.HasUniqueProperty(proTx.pubKeyOperator))
                 return _state.DoS(100, false, REJECT_CONFLICT, "bad-protx-dup-key");
 
-            auto dmn = std::make_shared<CDeterministicMN>(tx.GetHash(), proTx);
+            auto dmn = std::make_shared<CDeterministicMN>();
+            dmn->proTxHash = tx.GetHash();
+
+            // collateralOutpoint is either pointing to an external collateral or to the ProRegTx itself
+            if (proTx.collateralOutpoint.hash.IsNull()) {
+                dmn->collateralOutpoint = COutPoint(tx.GetHash(), proTx.collateralOutpoint.n);
+            } else {
+                dmn->collateralOutpoint = proTx.collateralOutpoint;
+            }
+
+            if (coinsView) {
+                Coin coin;
+                if (!coinsView->GetCoin(dmn->collateralOutpoint, coin) || coin.IsSpent() || coin.out.nValue != 1000 * COIN) {
+                    // should actually never get to this point as CheckProRegTx should have handled this case.
+                    // We do this additional check nevertheless to be 100% sure
+                    return _state.DoS(100, false, REJECT_INVALID, "bad-protx-collateral");
+                }
+                if (!proTx.collateralOutpoint.hash.IsNull() && (coin.nHeight == MEMPOOL_HEIGHT || pindexPrev->nHeight - coin.nHeight + 1 < 1)) {
+                    return _state.DoS(100, false, REJECT_DUPLICATE, "bad-protx-collateral-height");
```

Comment:
`REJECT_DUPLICATE`?

Reply:
Whoops, should have been `REJECT_INVALID`